src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -517,6 +533,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                                       const WORD scanCode,
                                       const ControlKeyStates modifiers,
                                       const bool keyDown)
+    {
+        auto keySentArgs = winrt::make<KeySentEventArgs>(vkey, scanCode, modifiers, keyDown);
+        _KeySentHandlers(*this, keySentArgs);

```

Comment:
Beware, this will cause a lot of needless IPC traffic.

Should reporting key events up to the app layer instead be the responsibility of `TermControl` (the last layer that knows about XAML?)

I know these are *not* xaml events, but handling it here has implications on the WPF control and eventually when Terminal is multi-process, will cause IPC amplification (hitting a key causes a synchronous (!) handler raise from Content Process -> Window Process, then a heap of synchronous (!) calls back down into `BroadcastKeyEvent` from Window Proc -> N Content Processes.)

Reply:
If I'm being completely honest, I don't think the control should be involved at all. Broadcasting is an application concern. It only "has" to be this way because we made key _bindings_ the control's concern.

Given that, though, we should restrict the damage across the layers as much as possible.